homeassistant/components/vacuum/__init__.py

```diff
@@ -245,45 +242,28 @@ def fan_speed_list(self):
         """Get the list of available fan speed steps of the vacuum cleaner."""
         raise NotImplementedError()
 
-    @property
-    def state_attributes(self):
-        """Return the state attributes of the vacuum cleaner."""
-        data = {}
-
-        if self.status is not None:
-            data[ATTR_STATUS] = self.status
-
-        if self.battery_level is not None:
-            data[ATTR_BATTERY_LEVEL] = self.battery_level
-            data[ATTR_BATTERY_ICON] = self.battery_icon
-
-        if self.fan_speed is not None:
-            data[ATTR_FAN_SPEED] = self.fan_speed
-            data[ATTR_FAN_SPEED_LIST] = self.fan_speed_list
-
-        return data
-
-    def turn_on(self, **kwargs):
-        """Turn the vacuum on and start cleaning."""
+    def start_pause(self, **kwargs):
+        """Start, pause or resume the cleaning task."""
         raise NotImplementedError()
 
-    def async_turn_on(self, **kwargs):
-        """Turn the vacuum on and start cleaning.
+    def async_start_pause(self, **kwargs):
+        """Start, pause or resume the cleaning task.
 
         This method must be run in the event loop and returns a coroutine.
         """
-        return self.hass.async_add_job(partial(self.turn_on, **kwargs))
+        return self.hass.async_add_executor_job(
```

Comment:
This method should not return anything. Instead `await` the task returned by `self.hass.async_add_executor_job`.

Reply:
Also update the docstring to reflect the change.